core/src/com/unciv/ui/worldscreen/mainmenu/WorldScreenOptionsPopup.kt

```diff
@@ -110,8 +111,13 @@ class WorldScreenOptionsPopup(val worldScreen:WorldScreen) : Popup(worldScreen)
         addYesNoRow ("Move units with a single tap", settings.singleTapMove) {
             settings.singleTapMove = it
         }
-        addYesNoRow ("Auto-assign city production", settings.autoAssignCityProduction) {
+        addYesNoRow ("Auto-assign city production", settings.autoAssignCityProduction, true) {
             settings.autoAssignCityProduction = it
+            if (it && UncivGame.Current.gameInfo.currentPlayerCiv.playerType == PlayerType.Human) {
```

Comment:
You didn't check if the current player is the viewing player. The proper check is something like `worldScreen.viewingCiv.isCurrentPlayer() && worldScreen.viewingCiv.playerType == PlayerType.Human`
Also this means that we don't access the UncivGame.Current, which is good - the less the better

Reply:
You're right I should have asked - not being able or willing to test multiplayer I had no idea how to do that check properly, just that some is probably needed. Sorry.